src/sync.js

```diff
@@ -824,10 +824,55 @@ SyncApi.prototype._sync = async function(syncOptions) {
 
 SyncApi.prototype._doSyncRequest = function(syncOptions, syncToken) {
     const qps = this._getSyncParams(syncOptions, syncToken);
+    console.log("[sync_dbg] Doing sync with query string: ", qps);
     return this.client._http.authedRequest(
         undefined, "GET", "/sync", qps, undefined,
         qps.timeout + BUFFER_PERIOD_MS,
-    );
+    ).then(r => {
+        // TODO: Remove logging once https://github.com/matrix-org/synapse/issues/7206
+        // is fixed
+
+        try {
+            if (localStorage && !localStorage.getItem("mx_do_sync_logging")) {
```

Comment:
Huh, somehow [tests are still failing](https://buildkite.com/matrix-dot-org/matrix-js-sdk/builds/1480#1d3c676c-59ea-46c9-998c-deb5c384ad5f) here with `ReferenceError: localStorage is not defined` at least on Buildkite... 🤔

Reply:
ugh, the environment throws an error if it's so much as referenced :/